src/Utils.ts

```diff
@@ -21,34 +22,67 @@ export class Utils {
      * @returns {Boolean} the result of the check of the Editor mode
      */
     public static isEditMode(): boolean {
-        return PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === "edit" ? true : false;
+        const aemMode = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE);
+        const param = new URL(document.location.href).searchParams.get(Constants.AEM_MODE);
+        const val = aemMode || param;
+        return val === Constants.AEM_EDIT;
     }
 
     /**
-     * Appends client lib to the page
+     * Appends javascript to the page
      */
-    public static appendClientLibs(clientLibUrl : string) {
+    public static appendScripts(clientLibUrl : string) {
         const script = document.createElement('script');
         const scripts = document.getElementsByTagName('script')[0];
         script.src = clientLibUrl;
         if (scripts.parentNode) {
             scripts.parentNode.insertBefore(script, scripts);
         }
     }
-
+    /**
+     * Appends stylesheets to the page
+     */
+    public static appendStyleSheets(clientLibUrl : string) {
+        const link = document.createElement('link');
+        const links = document.getElementsByTagName('link')[0];
+        link.href = clientLibUrl;
+        if (links.parentNode) {
+            links.parentNode.insertBefore(link, links);
+        }
+    }
 
     /**
      * Generates clientlib url
      */
-    public static generateClientLibsUrl(pageModelRoot : string, domain : string) : string {
-        let clientLibPath = '';
-        if (pageModelRoot != null) {
-            clientLibPath = domain + '/etc.clientlibs/' +  pageModelRoot.split("/")[2] + '/clientlibs/clientlib-base.min.js';
-        }
-        else {
-            throw new Error('Clientlib path cannot be determined! This should never happen.');
-            return "";
-        }
-        return clientLibPath;
+    public static generateClientLibsUrl(domain : string) : string[] {
+        const clientlibs: string[] =  Array<string>();
+        const path = '/etc.clientlibs/cq/gui/components/authoring/editors/clientlibs/internal/';
+        clientlibs.push(`${domain}${path}page.js`);
+        clientlibs.push(`${domain}${path}page.css`);
+        clientlibs.push(`${domain}${path}pagemodel/messaging.js`);
+        clientlibs.push(`${domain}${path}messaging.js`);
```

Comment:
Have you tried and plan to update the client library allowProxy config accordingly? Do you have a PoC?